02.calendar/calendar.rb

```diff
@@ -4,29 +4,28 @@
 require 'date'
 require 'optparse'
 
-def option_valid?
+def recieve_options
   params = ARGV.getopts('m:', 'y:')
-  option = { month: params['m'].to_i, year: params['y'].to_i }
-  option[:month] = Date.today.month unless (1..12).cover?(option[:month])
-  option[:year] = Date.today.year unless (1970..2100).cover?(option[:year])
-  option
+  recieved_options = { month: params['m'].to_i, year: params['y'].to_i }
```

Comment:
```suggestion
  options = { month: params['m'].to_i, year: params['y'].to_i }
```

これぐらいの短い名前でも十分かなと思いました。

Reply:
すみません、念のためここだけ認識が間違っていないか確認させてください🙇‍♂️！

レビューを頂くまで、メソッドの外と中や、違うメソッドの中同士など、スコープの範囲が被らない範囲でも同じ変数名を繰り返し使用することはNGと認識していました。(場合によっては意図していない不具合を招く原因になるから)
しかしそれは間違いで、「絶対にダメ！」ではなく、むしろ違うメソッド同士で中身が全く同じものを受け渡すような場合は可読性がよくなることもあり、そういう場合は変数名が重複しても問題ない。という認識で間違いないでしょうか？
(今回の場合、receive_optionsの中と、determine_datesの中でoptionsという同じ名前の変数を使用することになる。実質receive_optionsで渡したoptionsと、determine_datesで受け取るoptionsの中身は同じであり、同じ名前でも違和感はなく、むしろ無理に違う名前を付けるより同じ名前の方が読みやすい。)

Reply:
こちら、質問を見逃してました。回答するのでしばらくお待ちください 🙏

Reply:
> スコープの範囲が被らない範囲でも同じ変数名を繰り返し使用することはNGと認識していました。しかしそれは間違い
> むしろ違うメソッド同士で中身が全く同じものを受け渡すような場合は可読性がよくなる
> 同じ名前でも違和感はなく、むしろ無理に違う名前を付けるより同じ名前の方が読みやすい。

上記の認識でOKです 🙆‍♂️

> 場合によっては意図していない不具合を招く原因になるから

これ、具体例って挙げられます？スコープが被らないなら不具合も起きないのでは、と思ったのですが。

Reply:
回答ありがとうございます！🙏

> これ、具体例って挙げられます？

具体例、すみません、今自分では思いつきません…🙇‍♂️(学ぶうちにそういう場合が出てきてしまうのかな～と思っていた )

本で見たような記憶があったので見返したところ、チェリー本p110~111の"ブロックパラメータとブロック外の変数を同じ名前にする"パターンの解説あたりを勘違いしていたのかな…？と思いました。(もともと変数名を使いまわす→よくないみたいな印象はあった)
ブロックパラメータをブロック外の変数と同じ名前にした場合は最初の変数が機能しない→スコープが被っていないと勘違いして(ブロック内ではブロック外で定義した変数が使えるため、疑似的にスコープが被っていないように見えるだけで本当は被っている)
"スコープが被っていなくても、プログラム的に問題はなく動いても、勘違いしやすかったり意図せぬ不具合を招くから同じ名前はNGなんだ～"と勝手に拡大解釈して認識していたのかな…？？と思います。

認識は上述の質問内容でOKとのことで改めます！お手数おかけしました。

Reply:
なるほど、了解です 👌 
ちなみに変数のスコープは言語によって異なることもあるので、Ruby以外の言語を書くときはその点にも注意してみてください〜。